.github/workflows/gh-ci.yaml

```diff
@@ -43,7 +43,7 @@ jobs:
         matrix:
           # os: [macOS-latest, ubuntu-latest, windows-latest]
           os: [ubuntu-latest, macOS-latest]
-          python-version: ["3.9", "3.10", "3.11"]
+          python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
If you put an exclude for the combination of py3.12 + latest then this should work